src/python/pants/engine/process.py

```diff
@@ -316,21 +318,51 @@ def run(self, request: InteractiveProcess) -> InteractiveProcessResult:
 class BinaryPathRequest:
     search_path: Tuple[str, ...]
     binary_name: str
+    test_args: Optional[Tuple[str, ...]]
 
-    def __init__(self, *, search_path: Iterable[str], binary_name: str) -> None:
+    def __init__(
+        self,
+        *,
+        search_path: Iterable[str],
+        binary_name: str,
+        test_args: Optional[Iterable[str]] = None,
```

Comment:
This probably deserves some docstring. It looks like there are two things this should do, if implemented?

1. If it's an invalid program, return an error code != 0.
2. If valid, write to stdout a fingerprint of the program.

Reply:
Agreed.